### ISSUE TITLE
Maybe too many labels (30K+), it can not converge

### ISSUE THREAD

**Issue by lzh990711** [gy20073/compact_bilinear_pooling#10]
Thanks for your sharing!

I have now applying your code for clothing classification,  when I pretrain the network ( ft_last_layer.prototxt) using the script (ft_last_layer.sh ), and the number of training data is about 220K with ID 30K+, it can not converge. The initial loss is about 10, but after a lot of iterations about 100k+, the loss is still about 9. At the mean time , I have made another experiment, all the environment keeps the same as previous experiment except the number of training data become small(about 7K with 800ID) ,which is extracted from the previous training data, now  it converge quickly. 

 So my question is, when the training data is large, why does it not converged? are there existing some tricks to make its convergence?

Thank you very much!

**Comment by gy20073** [comment]
I haven’t tried to use the model on such a large number of classes. Right now, I have no idea why it doesn’t converge. 

Have you tried not using CBP to classify 30K classes? Will that converge? If that it doesn't, then I suspect the softmax layer might not be suitable for such a large number of classes. 

If otherwise, i.e. the usual softmax architecture converges, one thing you could try is to increase the number of projection dimensions(num_output: 4096) to something bigger like 4096*16. 

Let me know your results and we could continue to figure out why this happens!

**Comment by lzh990711** [comment]
Thanks for your reply!
I have ever trained the GoogleNet-v2 by using the training data  about 220K with ID 30K+, which finally can converge.  Now according to your suggestion, the experiment is under training, if there has any development, I will give your answer right now!

**Comment by lzh990711** [comment]
I have made some attempts according to your suggestion, but the result is unsatisfactory, it can not converge if the number of labels is more than 30K ID.

**Comment by gy20073** [comment]
Can you upload your experiment scripts? Including the GoogleNet one working on 30K ID, and the ones that CBP working on 7K ID and not working on 30K ID.

**Comment by billhhh** [comment]
> increase the number of projection dimensions(num_output: 4096) to something bigger like 4096*16.

is that means the num_output of compact_bilinear_param {
    num_output: 8192
    sum_pool: true
  }？

**Comment by gy20073** [comment]
yes

**Comment by billhhh** [comment]
thank you, gy20073!
